app/helpers/common_helper.rb

```diff
@@ -11,6 +11,12 @@ def render_google_tag_manager_head_code
     render partial: "common/google_tag_manager_head" if SiteSetting.gtm_container_id.present?
   end
 
+  def render_csp_nonce_code
+    if SiteSetting.content_security_policy || SiteSetting.content_security_policy_report_only
+      render partial: "common/csp_nonce"
```

Comment:
In CSP, using an `nonce` has one side-effect that is tricky for our use case, it will invalidate any `unsafe-inline` or `unsafe-eval` entries. We recommend not using these, of course, but in some cases they are inevitable, and we'd cause a regression by always having a `nonce` included in the CSP directives. 

See https://csp.withgoogle.com/docs/strict-csp.html#:~:text=Note%3A%20In%20the%20presence%20of,allow%20inline%20scripts%20to%20execute. 

I think in this case, one workaround may be to add a second check here for: 

```
if SiteSetting.gtm_container_id || SiteSetting.content_security_policy_include_script_src_nonce
```

Where `gtm_container_id` ensures current GTM users work fine and `content_security_policy_include_script_src_nonce` is a hidden site setting that instances can flip to include the nonce at all times.